engine/src/main/java/org/terasology/engine/rendering/nui/CoreScreenLayer.java

```diff
@@ -80,6 +80,12 @@ public void setDepthAuto() {
         }
     }
 
+    @Override
+    @SuppressWarnings("PMD.UselessOverridingMethod") // override elevates access modifier
```

Comment:
what does "elevates access modifier" mean in this context, resp why one want to have it here?

Reply:
The method is only calling its super and thus wouldn't be required, _but_ the super method is `protected` while the overriding method is `public`, thereby elevating the accessibility of the method. Without the override, package-external entities would not be able to call the (super) method.

If we look at the jenkins build error in https://jenkins.terasology.io/job/Terasology/job/engine/job/PR-5149/3/console that's exactly the problem we see:
```
/home/jenkins/agent/workspace/Terasology_engine_PR-5149/engine/src/main/java/org/terasology/engine/rendering/nui/internal/NUIManagerInternal.java:421: error: setId(String) has protected access in AbstractWidget
        screen.setId(uri.toString());
              ^
```

`NUIManagerInternal` needs to call this method but is not able to without the override that allows the method to be called by package-external entities. Hence we need the override as it elevates the access modifier.

Note: Whether `NUIManagerInternal` should call this method or whether the super method should be public in the first place is a valid question but it goes beyond this refactoring here and I consciously decide to ignore this potential rabbit hole for now.

Reply:
I think it can make sense to have some methods protected on an interface/super class, and exposing them in sub-classes. For instance, if implementations are not in the same package but are supposed to handle internals. 

Whether or not it makes sense here in particular is something we could double check again.